File: solve/10845.cs
using System.Text;
var sb = new StringBuilder();
var q = new Queue<int>();
var n = int.Parse(Console.ReadLine());
var back = -1;
while (n-->0)
{
    var spl = Console.ReadLine().Split();
    switch(spl[0])
    {
        case "push":
            back = int.Parse(spl[1]);
            q.Enqueue(back);
            break;
        case "pop":
            if (q.TryDequeue(out int v1))
                sb.AppendLine(v1.ToString());
            else
                sb.AppendLine("-1");
            break;
        case "size":
            sb.AppendLine(q.Count.ToString());
            break;
        case "empty":
            sb.AppendLine(q.Count == 0 ? "1" : "0");
            break;
        case "front":
            if (q.TryPeek(out int v2))
                sb.AppendLine(v2.ToString());
            else
                sb.AppendLine("-1");
            break;
        case "back":
            if (q.Count == 0)
                sb.AppendLine("-1");
            else
                sb.AppendLine(back.ToString());
            break;
    }
}
Console.WriteLine(sb.ToString());

File: solve/10845.cpp
#include <iostream>
#include <cstdio>
#include <algorithm>
#include <queue>
#include <string>

using namespace std;

int main() 
{
    ios::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);

    queue<int> q;
    int n, v;
    string cmd;
    cin >> n;

    for (int i = 0; i < n; i++)
    {
        cin >> cmd;

        if (cmd == "push")
        {
            cin >> v;
            q.push(v);
        }
        else if (cmd == "pop")
        {
            if (q.empty()) cout << "-1\n";
            else 
            {
                cout << q.front() << "\n"; 
                q.pop();
            }
        }
        else if (cmd == "size")
        {
            cout << q.size() << "\n";
        }
        else if (cmd == "empty")
        {
            cout << (q.empty() ? "1" : "0") << "\n";
        }
        else if (cmd == "front")
        {
            if (q.empty()) cout << "-1\n";
            else cout << q.front() << "\n";
        }
        else if (cmd == "back")
        {
            if (q.empty()) cout << "-1\n";
            else cout << q.back() << "\n";
        }
    }

    return 0;
}